File: atcoder_cpp/ABC_173-C_20200707.cpp
#include <bits/stdc++.h>
#define rep(i, n) for (int i = 0; i < (int)(n); i++)
using namespace std;
using ll = long long;
using P = pair<int, int>;

int main() {
    int H, W, K;
    cin >> H >> W >> K;
    vector<string> C(H);
    rep(i, H) cin >> C.at(i);
    int ans = 0;
    rep(i, 1<<H) rep(j, 1<<W) {
        int cnt = 0;
        rep(m, H) rep(n, W){
            if(i>>m&1) continue;
            if(j>>n&1) continue;
            if(C[m][n] == '#') cnt++;
        }
        if (cnt == K) ans++;
    }
    cout << ans << endl;
}

File: atcoder_py/ABC_173-C_20200707.py
H, W, K = map(int, input().split())
C = []
for _ in range(H): C.append([c for c in input()])
ans = 0
for i in range(1<<H):
    for j in range(1<<W):
        cnt = 0
        for m in range(H):
            for n in range(W):
                if i>>m&1: continue
                if j>>n&1: continue
                if C[m][n] == '#': cnt += 1
        if cnt == K: ans += 1
print(ans)